src/debounce/index.ts

```diff
@@ -106,21 +122,29 @@ export function debounce<T>(
   });
 
   sample({
-    source: $timeout,
+    source: { timeout: $timeout, canceller: $canceller },
     clock: triggerTick,
-    target: timerFx,
+    fn: ({ timeout, canceller }) => ({ timeout, canceller: canceller ?? {} }),
+    target: spread({
```

Comment:
Need to check, that does not lead to `spread` duplicate in the bundle 🤔 

If it does - simplest option would be to just split it into two `sample`, i think